src/mame/wyse/wy50.cpp

```diff
@@ -320,6 +337,9 @@ void wy50_state::wy50(machine_config &config)
 	modem.rxd_handler().set(m_sio, FUNC(scn2661b_device::rxd_w));
 	modem.cts_handler().set(m_sio, FUNC(scn2661b_device::cts_w));
 	modem.dcd_handler().set(m_sio, FUNC(scn2661b_device::dcd_w));
+
+	SPEAKER(config, "speaker").front_center();
+	BEEP(config, m_beep, 950).add_route(ALL_OUTPUTS, "speaker", 0.10);  // Star Micronics QMB06 PZT Buzzer
```

Comment:
You can find specifications for this device trivially: https://www.electronicsurplus.com/star-qmb06-audio-alarms-and-buzzers-pzt-sound-transducer-buzzer

Reply:
For some reason the frequency 2048 sounds too high.  The only video I could find on youtube with the beep sounds is this one, right at the beginning of the video and at the 9:32 mark:

https://www.youtube.com/watch?v=xQTr9ZOJkC0
https://youtu.be/xQTr9ZOJkC0?t=570

Reply:
QMB06 datasheet says 2048hz, are you sure that's the buzzer on this WY50/75?
Try 1024hz then, that's more close to what that video sounds like.

Reply:
Looking at the maintentance manual:

https://doc.lagout.org/science/0_Computer Science/0_Computer History/old-hardware/wyse/WY-50/Wyse_WY-50_Terminal_Mainenance_198601.pdf

theres a bunch of schematics in the pdf manual of different versions, p213 for example has the STAR QMB06 shown in a circuit that has a bunch of resistors and capacitors and 74LS14s and diodes along with an LED.  I wonder if that affects the sound of the buzzer.

Reply:
<a href="https://imgur.com/MuxYEQY"><img src="https://i.imgur.com/MuxYEQY.png" title="source: imgur.com" /></a>

Reply:
Looking at the datasheet it looks like there's a bunch of resonant frequency peaks, if I read it right, one about 660hz too.

<a href="https://imgur.com/0YhREwV"><img src="https://i.imgur.com/0YhREwV.png" title="source: imgur.com" /></a>

        SPEAKER(config, "speaker").front_center();
        BEEP(config, m_beep,  2048).add_route(ALL_OUTPUTS, "speaker", 0.10);  // Star Micronics QMB06 PZT Buzzer
        BEEP(config, m_beep2,  680).add_route(ALL_OUTPUTS, "speaker", 0.05);  // Star Micronics QMB06 PZT Buzzer
        BEEP(config, m_beep3, 4500).add_route(ALL_OUTPUTS, "speaker", 0.05);  // Star Micronics QMB06 PZT Buzzer
        BEEP(config, m_beep4, 8800).add_route(ALL_OUTPUTS, "speaker", 0.025);  // Star Micronics QMB06 PZT Buzzer

        m_beep->set_state(BIT(data, 5));
        m_beep2->set_state(BIT(data, 5));
        m_beep3->set_state(BIT(data, 5));
        m_beep4->set_state(BIT(data, 5));


If I make multiple beep devices for each one of the peaks, it has a different sound, it makes it sound lower than 2048 hz.

Maybe 4 different beepers is overkill, but having 2 (at 2048hz and 680hz) sounds better than just the single beeper at 2048hz.

Reply:
I see, that looks like it's an LC filter. See the coil, and then the cap to GND.
Multiple beepers? nah, I wouldn't do that.

SPEAKER(config, "speaker").front_center();
// Star Micronics QMB06 PZT Buzzer (2048Hz peak) + LC filter, output frequency is approximated here
BEEP(config, m_beep, 1000).add_route(ALL_OUTPUTS, "speaker", 0.10);

Reply:
ok, I put the single beeper back with your LC comment and 1000 hz beeper.

SPEAKER(config, "speaker").front_center();
// Star Micronics QMB06 PZT Buzzer (2048Hz peak) + LC filter, output frequency is approximated here
BEEP(config, m_beep, 1000).add_route(ALL_OUTPUTS, "speaker", 0